src/main/java/efs/task/syntax/GuessNumberGame.java

```diff
@@ -13,10 +18,66 @@ public static void main(String[] args) {
     }
 
     public GuessNumberGame(String argument) {
-        //TODO: Implement the constructor
+        try {
+            this.maxNumber = Integer.parseInt(argument);
+        } catch (NumberFormatException e) {
+            System.out.println("'" + argument + "'" + " to " +UsefulConstants.WRONG_ARGUMENT+ " - to nie liczba");
+            throw new IllegalArgumentException(e);
+        }
+
+        if (maxNumber < 1 || maxNumber > 400) {
+            System.out.println(argument + " to " +UsefulConstants.WRONG_ARGUMENT+ " - jest spoza zakresu <1," +UsefulConstants.MAX_UPPER_BOUND+ ">");
+            throw new IllegalArgumentException();
+        }
+
+        this.numberToGuess = (int) (Math.log(maxNumber) / Math.log(2)) + 1;
+        this.maxTries = Math.floor(Math.log((double) this.maxNumber) / Math.log(2)) + 1;
```

Comment:
_clean code_ wydawałoby się, że _maxTries_ powinno przechowywać całkowitą liczbę, więc może warto to zrzutować już tutaj?